tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,314 @@
+"""
+==========================
+Arranging Axes in a Figure
+==========================
+
+Often more than one axes is wanted on a figure at a time, and usually
+we want to organize those axes into a regular grid.  Matplotlib has a
+variety of tools for working with grids of axes that have evolved over
+the history of the library.  Here we will discuss the tools we think
+users should use most often, and then dicuss some of the older tools,
+and the tools that underpin how axes are organized.
+
+How to create grid-shaped combinations of axes:
```

Comment:
I understand the motivation to give an overview in terms of taxonometry here, but that introduces, quite some overhead (GridSpec, SubplotSpec, subplot2grid) that is not relevant for people only needing relatively simple axes arrangements. IMHO the tutorial should focus more on practical usage. If you want to keep this, could this whole block be moved to the bottom and maybe only linked with a single sentence here?

Reply:
Again, if we were starting fresh and we could ask our users to only read our docs, I'd totally agree with you.  However, I think we are fighting against >30 years of history of `subplot` being how one makes a grid of axes, and >15 years of Matplotlib where `gridspec` (etc) was the documented way of making more complicated arrangements. Given that history, I think it is more confusing to _not_ mention them.

Reply:
So I went the other way for now, and explicitly discuss `subplot` and `subplot2grid` at the top.  I also organized this top section with section and subsection to hopefully make it easier to parse.  I also tried to make ti clear why we were not covering these in this tutorial.